Chronological/Filter.cs

```diff
@@ -201,6 +201,15 @@ private static string MemberExpressionToString(MemberExpression memberExpression
             throw new NotImplementedException();
         }
 
+        private static ExpressionType GetInnermostExpressionType(MemberExpression memberExpression)
```

Comment:
This works great for anonymous types but not for normal types, it fails with a null reference on the innermost value because it doesn't have an Expression. I think you can fix with:

```cs
        private static ExpressionType GetInnermostExpressionType(MemberExpression memberExpression)
        {
            if (memberExpression.Expression == null)
            {
                return memberExpression.NodeType;
            }
            if (memberExpression.Expression.NodeType == ExpressionType.MemberAccess)
            {
                return GetInnermostExpressionType(memberExpression.Expression as MemberExpression);
            }
            return memberExpression.Expression.NodeType;
        }
```